src/EditorFeatures/Test2/IntelliSense/CSharpSignatureHelpCommandHandlerTests.vb

```diff
@@ -202,14 +202,22 @@ class Program
 }
 ]]></Document>)
 
+                ' We don't have a definite symbol, so default to first
                 state.SendTypeChars("(")
                 Await state.AssertSignatureHelpSession()
                 Await state.AssertSelectedSignatureHelpItem("void Program.F(int i)")
                 Assert.Equal(2, state.CurrentSignatureHelpPresenterSession.SignatureHelpItems.Count)
 
-                state.SendTypeChars(""""",")
+                ' We now have a definite symbol (the string overload)
+                state.SendTypeChars("""""")
                 Await state.AssertSignatureHelpSession()
-                Await state.AssertSelectedSignatureHelpItem("void Program.F(int i)")
+                Await state.AssertSelectedSignatureHelpItem("void Program.F(string s)")
+                Assert.Equal(2, state.CurrentSignatureHelpPresenterSession.SignatureHelpItems.Count)
+
+                ' We don't have a definite symbol again, but we stick to our last choice
```

Comment:
> We don't have a definite symbol again

❓ Do we have any way to assert this statement?

Reply:
Hum, it doesn't seem straightforward. I think it would require storing it in `Model`, but that doesn't seem useful if for testing only.

The key logic is in `Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.SignatureHelp.Controller.Session.ComputeModelInBackgroundAsync`:
```C#
// this picks `items.SelectedItemIndex.HasValue`, the last valid selection, or defaults
var selectedItem = GetSelectedItem(currentModel, items, provider); 

// after that we use different parts of `items` and also `selectedItem`, to build a `Model`, but `items.SelectedItemIndex.HasValue` is ignored after that
```